code/modules/mob/living/carbon/hallucination.dm

```diff
@@ -0,0 +1,1102 @@
+
+
+
+/datum/hallucination/insides
+	duration = 60
+	max_power = 75
+
+/datum/hallucination/insides/start()
+	if(ishuman(holder))
+		var/mob/living/carbon/human/H = holder
+		var/obj/O = pick(H.organs)
+		to_chat(holder,SPAN_DANGER("You feel something [pick("moving","squirming","skittering", "writhing", "burrowing", "crawling")] inside of your [O.name]!"))
+	else
+		to_chat(holder,SPAN_DANGER("You feel something [pick("moving","squirming","skittering", "writhing", "burrowing", "crawling")] inside of you!"))
+	if(prob(min(holder.hallucination/2, 80)))
+		sound_to(holder, pick('sound/misc/zapsplat/chitter1.ogg', 'sound/misc/zapsplat/chitter2.ogg', 'sound/effects/squelch1.ogg', 'sound/effects/lingextends.ogg'))
+
+/datum/hallucination/insides/end()
+	if(prob(50))
+		to_chat(holder, SPAN_WARNING(pick("You see something moving under your skin!", "Whatever it is, it's definitely alive!", "If you don't get it out soon...", "It's moving towards your mouth!")))
+	..()
+
+
+
+/datum/hallucination/pain				//Pain. Picks a random type of pain, and severity is based on their level of hallucination.
+/datum/hallucination/pain/start()
+	var/pain_type = rand(1,5)
+	switch(pain_type)
+		if(1)
+			to_chat(holder,SPAN_DANGER("You feel a sharp pain in your head!"))
+		if(2)
+			switch(holder.hallucination)
+				if(1 to 15)
+					to_chat(holder, SPAN_WARNING("You feel a light pain in your head."))
+				if(16 to 40)
+					to_chat(holder, SPAN_DANGER("You feel a throbbing pain in your head!"))
+				if(41 to INFINITY)
+					to_chat(holder, SPAN_DANGER("You feel an excruciating pain in your head!"))
+					holder.emote("me",1,"winces.")
+					holder.eye_blurry += 9
+		if(3)
+			switch(holder.hallucination)
+				if(1 to 15)
+					to_chat(holder, SPAN_WARNING("The muscles in your body hurt a little."))
+				if(16 to 40)
+					to_chat(holder, SPAN_DANGER("The muscles in your body cramp up painfully."))
+				if(41 to INFINITY)
+					to_chat(holder, SPAN_DANGER("There's pain all over your body."))
+					holder.emote("me",1,"flinches as all the muscles in their body cramp up.")
+					holder.eye_blurry += 9
+		if(4)
+			switch(holder.hallucination)
+				if(1 to 15)
+					to_chat(holder, SPAN_WARNING("You feel a slight itch."))
+				if(16 to 40)
+					to_chat(holder, SPAN_DANGER("You want to scratch your itch badly!"))
+				if(41 to INFINITY)
+					to_chat(holder, SPAN_DANGER("This itch makes it really hard to concentrate!"))
+					holder.emote("me",1,"shivers slightly.")
+					holder.eye_blurry += 9
+		if(5)
+			switch(holder.hallucination)
+				if(1 to 15)
+					to_chat(holder, SPAN_WARNING("You feel a little too warm."))
+				if(16 to 40)
+					to_chat(holder, SPAN_DANGER("You feel a horrible burning sensation!"))
+				if(41 to INFINITY)
+					to_chat(holder, SPAN_DANGER("It feels like you're being burnt to the bone!"))
+					holder.emote("me",1,"flinches.")
+					holder.eye_blurry += 9
+
+	holder.adjustHalLoss(min(holder.hallucination / 3, 25))		//always cause fake pain
```

Comment:
Consider making this pick a limb then directly adding pain to that limb. `/obj/item/organ/external/proc/add_pain`